### PR TITLE
Add Connector#getMaterializedViewProperties

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

**Comment by sopel39** [comment]
@anjalinorwood could you review Iceberg commit

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java [hunk 1]
```diff
@@ -52,6 +54,32 @@ public void setUp()
         assertQuery("SELECT count(*) FROM base_table2", "VALUES 3");
     }
 
+    @Test
+    public void testShowCreate()
+    {
+        assertUpdate("CREATE MATERIALIZED VIEW materialized_view_with_property " +
+                "WITH (partitioning = ARRAY['_date']) AS " +
+                "SELECT _bigint, _date FROM base_table1");
+        assertQuery("SELECT COUNT(*) FROM materialized_view_with_property", "VALUES 6");
```
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:64]: isn't that tested somewhere else?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java [hunk 2]
```diff
@@ -52,6 +54,32 @@ public void setUp()
...
+                                "WITH (\n" +
+                                "   format = 'ORC',\n" +
+                                "   partitioning = ARRAY['_date']\n" +
+                                ") AS\n" +
+                                "SELECT\n" +
+                                "  _bigint\n" +
+                                ", _date\n" +
+                                "FROM\n" +
+                                "  base_table1");
+        assertThatThrownBy(() -> computeActual("CREATE MATERIALIZED VIEW materialized_view_with_property " +
```
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:77]: this deserves another test method

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java [hunk 3]
```diff
@@ -80,6 +81,13 @@ public void testShowCreate()
         assertUpdate("DROP MATERIALIZED VIEW materialized_view_with_property");
     }
 
+    @Test
+    public void testSystemMaterializedViewProperties()
```
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:85]: no unit tests in the engine itself?

**Review by sopel39 — COMMENTED** [review]
lgtm % small comments

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java (hunk 1 above)
Comment (raunaqmorarka) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:64]: All the other tests in this class are disabled due to https://github.com/trinodb/trino/issues/5892, until that's fixed it's useful to test a bit extra here.

File: core/trino-main/src/main/java/io/trino/execution/CreateMaterializedViewTask.java [hunk 4]
```diff
@@ -105,7 +105,7 @@ public String explain(CreateMaterializedView statement, List<Expression> paramet
                 .orElseThrow(() -> new TrinoException(NOT_FOUND, "Catalog does not exist: " + name.getCatalogName()));
 
         Map<String, Expression> sqlProperties = mapFromProperties(statement.getProperties());
-        Map<String, Object> properties = metadata.getTablePropertyManager().getProperties(
+        Map<String, Object> properties = metadata.getMaterializedViewPropertyManager().getProperties(
```
[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/execution/CreateMaterializedViewTask.java:108]: Added TestCreateMaterializedViewTask

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java (hunk 3 above)
Comment (raunaqmorarka) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:90]: Yes, I didn't find unit tests for other system.metadata tables either.
There are product tests covering it though.

File: core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+                tablePropertyManager,
+                materializedViewPropertyManager,
+                testCatalog.getConnectorCatalogName());
+        parser = new SqlParser();
+        statsCalculator = createNewStatsCalculator(metadata, new TypeAnalyzer(parser, metadata));
+        queryStateMachine = stateMachine(transactionManager, createTestMetadataManager(), new AllowAllAccessControl());
+    }
+
+    @Test
+    public void testCreateMaterializedNotExistsTrue()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java:129]: `testCreateMaterializedViewIfNotExists`

File: core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+                ImmutableList.of(),
+                Optional.empty());
+
+        getFutureValue(new CreateMaterializedViewTask(parser, new TestingGroupProvider(), statsCalculator)
+                .execute(statement, transactionManager, metadata, new AllowAllAccessControl(), queryStateMachine, ImmutableList.of(), WarningCollector.NOOP));
+        assertEquals(metadata.getCreateMaterializedViewCallCount(), 1);
+    }
+
+    @Test
+    public void testCreateMaterializedNotExistsFalse()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java:146]: `testCreateMaterializedViewWithExistingView`

File: core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+        assertTrinoExceptionThrownBy(() -> getFutureValue(new CreateMaterializedViewTask(parser, new TestingGroupProvider(), statsCalculator)
+                .execute(statement, transactionManager, metadata, new AllowAllAccessControl(), queryStateMachine, ImmutableList.of(), WarningCollector.NOOP)))
+                .hasErrorCode(ALREADY_EXISTS)
+                .hasMessage("Materialized view already exists");
+
+        assertEquals(metadata.getCreateMaterializedViewCallCount(), 1);
+    }
+
+    @Test
+    public void testCreateMaterializedViewWithTablePropertyFails()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java:166]: `testCreateMaterializedViewWithInvalidProperty`

File: core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+            ImmutableMap.of("baz", "property_value"));
+
+    private Session testSession;
+    private MockMetadata metadata;
+    private TransactionManager transactionManager;
+    private SqlParser parser;
+    private StatsCalculator statsCalculator;
+    private QueryStateMachine queryStateMachine;
+
+    @BeforeMethod
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java:101]: Is it possible to add successful case where view is actually created (with MV property)?

**Review by sopel39 — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java (hunk 8 above)
Comment (raunaqmorarka) [reply core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java:101]: For successful case with MV property it ends up trying to cast a varchar(3) to varchar and fails due to missing cast function in the testing metadata manager created by createTestMetadataManager.
I have this in the iceberg tests

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java (hunk 1 above)
Comment (anjalinorwood) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java:73]: The tests are disabled due to an unrelated issue. In any case the fix for that issue and re-enabling of tests is here: https://github.com/trinodb/trino/pull/5942/commits. Would be great to get that PR in .. another person in the community reached out to me saying they were hitting that issue as well. @sopel39 @raunaqmorarka

**Review by anjalinorwood — COMMENTED** [review]
lgtm.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConnector.java [hunk 9]
```diff
@@ -170,6 +170,12 @@ public List<PropertyMetadata<?>> getTableProperties()
         return tableProperties;
     }
 
+    @Override
+    public List<PropertyMetadata<?>> getMaterializedViewProperties()
+    {
+        return tableProperties;
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConnector.java:176]: This should have a code comment explaining why MV properties = table properties

**Pull request merged**